src/ICSharpCode.SharpZipLib/Zip/FastZip.cs

```diff
@@ -602,7 +602,13 @@ void ExtractEntry(ZipEntry entry)
 			if (doExtraction && !Directory.Exists(dirName)) {
 				if (!entry.IsDirectory || CreateEmptyDirectories) {
 					try {
-						Directory.CreateDirectory(dirName);
+						if (continueRunning_ = events_.OnProcessDirectory(dirName, true)) {
```

Comment:
`events_` is not checked for null. This is probably what causes the tests to fail.

Could be solved with:
```.cs
if (continueRunning_ = events_?.OnProcessDirectory(dirName, true) ?? true) {
```